ha_spider.cc

```diff
@@ -7472,6 +7474,29 @@ int ha_spider::rnd_next_internal(
 #ifdef WITH_PARTITION_STORAGE_ENGINE
     check_select_column(TRUE);
 #endif
+    if (this->result_list.direct_limit_offset)
+    {
+      if (this->trx->thd->select_limit == 0)
+      { // mean has got all result
+        DBUG_RETURN(check_error_mode_eof(HA_ERR_END_OF_FILE));
+      }
+      longlong table_count = this->records();
+      if (table_count <= this->trx->thd->select_offset)
+      {
+        // skip this spider(partition)
+        this->trx->thd->select_offset -= table_count;
+        DBUG_RETURN(check_error_mode_eof(HA_ERR_END_OF_FILE));
```

Comment:
If you need a lot of time for thinking about it, I can do it after merging this changes.